api/step/plan.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2023 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package step
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/go-vela/server/database"
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/library"
+	"github.com/go-vela/types/pipeline"
+)
+
+// PlanSteps is a helper function to plan all steps
+// in the build for execution. This creates the steps
+// for the build in the configured backend.
+func PlanSteps(database database.Interface, p *pipeline.Build, b *library.Build) ([]*library.Step, error) {
+	// variable to store planned steps
+	steps := []*library.Step{}
+
+	// iterate through all pipeline stages
+	for _, stage := range p.Stages {
+		// iterate through all steps for each pipeline stage
+		for _, step := range stage.Steps {
+			// create the step object
+			s := new(library.Step)
```

Comment:
What would you think of refactoring this file just a bit farther to reduce duplication a bit?

I think it would be nice to add a `planStep` function, and then update the `PlanSteps` function so it looks like this:

```go
steps := []*library.Step{}

for _, stage := range p.Stages {
	for _, step := range stage.Steps {
		steps = append(steps, planStep(step))
	}
}

for _, step := range p.Steps {
	steps = append(steps, planStep(step))
}
```